src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -122,7 +123,7 @@ internal Win32FileSystemEnumerableIterator(string path, string originalUserPath,
             Debug.Assert(searchOption == SearchOption.AllDirectories || searchOption == SearchOption.TopDirectoryOnly);
             Debug.Assert(resultHandler != null);
 
-            _oldMode = Interop.Kernel32.SetErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS);
+            _setBackOldMode = Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS, out _oldMode);
```

Comment:
Sounds reasonable. Seems like something that's more app policy than something the framework should be doing, anyway, and msdn suggests it's a best practice for all apps to set this. The downside would seem to be if a host doesn't set it and the lack of it then causes problems, but I don't know how much of an actual problem that is rather than theoretical.